gnomad_qc/v4/assessment/calculate_per_sample_stats.py

```diff
@@ -426,21 +458,50 @@ def main(args):
         tmp_dir="gs://gnomad-tmp-30day",
     )
     # SSA Logs are easier to troubleshoot with.
-    hl._set_flags(use_ssa_logs="1")
+    hl._set_flags(use_ssa_logs="1", no_whole_stage_codegen="1")
 
     data_type = args.data_type
-    test = args.test
+    autosomes_only = args.autosomes_only
+    test_dataset = args.test_dataset
+    test_partitions = (
+        list(range(args.test_n_partitions)) if args.test_n_partitions else None
+    )
+
+    # The following four exome partitions have given us trouble with out-of-memory
+    # errors in the past. We are testing on these partitions to ensure that the
+    # per-sample stats script can handle them before running the entire dataset.
+    if data_type == "genomes" and args.test_difficult_partitions:
+        raise ValueError(
+            "Difficult partitions for testing have only been chosen for exomes data."
+        )
+
+    test_partitions = (
```

Comment:
Does this test require a different cluster setup?

Reply:
It requires the changes coming in the next PR